dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/SerializationInfoCache.java

```diff
@@ -0,0 +1,569 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.session.store.cache;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.ibm.ws.session.store.cache.TypeConversion;
+
+public class SerializationInfoCache {    
+
+    public static BuiltinSerializationInfo<?> lookupByClass(Class<?> c) {
+        return BUILTIN_MAPPINGS.get(c);
+    }
+
+    public static BuiltinSerializationInfo<?> lookupByIndex(short index) {
```

Comment:
It seems the way you are using it, instead of short it should be a byte. Same with all the constants LONG, INTEGER, BUILTIN_SERIALIZATION, etc and the return value from getIndex();